Makes clustercontroller-core work on WiFi

hakonhall opened the pull request.

Commit:
Makes clustercontroller-core work on WiFi

vekterli reviewed (approved):
LG for cluster controller changes, but since this touches JRT I'll loop in @havardpe for those changes.

havardpe reviewed (commented):
JRT changes seems to complexify things and will also (as far as I can see) break 'everything'.

Commit:
Improve Spec API

 - Removes Spec.getLocalHostName
 - Removes distinction between listening- and connect- address for Spec
 - Makes all usage of connect w/Spec specify hostname

havardpe reviewed (commented):
I suggest keeping the changes in jrt.Spec to the comment on the constructor taking only an int and removing the HostName import. Other changes seems to be unintentional left-overs from previous iterations of the code.

otherwise it looks good.

Commit:
Properly revert most changes to Spec

Commit:
Remove listening

Commit:
Restore port doc
